### PR TITLE
Clean up 7 small functions in player.cpp

### PR DESCRIPTION
**Pull request by AJenbo**
SetPlayerOld PlrHitObj PlrDeathModeOK CheckCheatStats ClrPlrPath and
SyncInitPlr are now bin exact

Further clean up ShieldDur

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -3514,20 +3511,22 @@ bool __fastcall PlrHitPlr(int pnum, char p)
...
-bool __fastcall PlrHitObj(int pnum, int mx, int my)
+BOOL __fastcall PlrHitObj(int pnum, int mx, int my)
 {
-	int oi; // edx
+	int oi;
 
-	if ( dObject[mx][my] <= 0 )
-		oi = -1 - dObject[mx][my];
-	else
+	if ( dObject[mx][my] > 0 ) {
```
Comment (seritools) [inline Source/player.cpp]: I'm guessing this whole if/else can be replaced by `abs(dObject[mx][my]) - 1`.
Could you check that? Either way I'd change the else branch to `oi = -dObject[mx][my] - 1;`

File: Source/player.cpp [hunk 2]
```diff
@@ -4689,33 +4691,38 @@ void __cdecl ProcessPlayers()
...
-	int v1; // esi
-
-	v1 = pnum;
-	if ( (unsigned int)pnum >= MAX_PLRS )
+	if ( (DWORD)pnum >= MAX_PLRS ) {
 		TermMsg("ClrPlrPath: illegal player %d", pnum);
-	memset(plr[v1].walkpath, -1, 0x19u);
+	}
+
+	memset(plr[pnum].walkpath, -1, 0x19u);
```
Comment (seritools) [inline Source/player.cpp]: Did you try replacing that memset with a simple for loop?

File: Source/player.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/player.cpp]: abs() and other odd variations didn't produce the same asm, but -dObject is a good point.

File: Source/player.cpp (hunk 2 above)
Comment (AJenbo) [reply Source/player.cpp]: Didn't work, but changed it to sizeof(plr[pnum].walkpath) instead of 0x19u :)

**Pull request merged**